src/backend/aspen/app/views/sample.py

```diff
@@ -531,31 +534,52 @@ def update_sample_public_ids():
             f"Private Identifiers {missing_private_identifiers} not found in DB",
         )
 
-    # check that public_identifiers don't already exist
-    existing_public_ids: list[str] = api_utils.get_existing_public_ids(
-        request_public_ids, g.db_session, group_id=group_id
+    samples_to_update = g.db_session.query(Sample).filter(
+        and_(
+            Sample.submitting_group_id == group_id,
+            Sample.private_identifier.in_(private_to_public.keys()),
+        )
     )
-    if existing_public_ids:
-        raise ex.BadRequestException(
-            f"Public Identifiers {existing_public_ids} are already in the database",
+
+    if public_ids_are_gisaid_isl:
+        samples_to_update = samples_to_update.options(
+            joinedload(Sample.uploaded_pathogen_genome),
         )
+        for s in samples_to_update:
+            isl_number = private_to_public[s.private_identifier]
+            accessions = s.uploaded_pathogen_genome.accessions()
+
+            if accessions:
+                for accession in accessions:
+                    if isinstance(accession, GisaidAccession):
+                        accession.public_identifier = isl_number
+            else:
+                # create a new accession if DNE
+                s.uploaded_pathogen_genome.add_accession(
+                    repository_type=PublicRepositoryType.GISAID,
+                    public_identifier=isl_number,
+                    workflow_start_datetime=datetime.datetime.now(),
+                    workflow_end_datetime=datetime.datetime.now(),
+                )
```

Comment:
are we missing some calls to `g.db_session.add()` and `g.db_session.commit()` here, or is that implicit?

Reply:
it was working without adding those calls in the test, but i added them now anyways